### DISCUSSION TITLE
Using Skia with RenderTargetBitmap

### DISCUSSION THREAD

**Discussion by amwx** [AvaloniaUI/Avalonia#11599] [Q&A]
Had code that was doing some custom Skia drawing into a RenderTargetBitmap that's no longer compatible with the recent DrawingContext refactoring:
```C#
using var dc = (_tempBitmap as RenderTargetBitmap).CreateDrawingContext();
var leaseFeature = dc.GetFeature<ISkiaSharpApiLeaseFeature>();
if (leaseFeature == null)
    throw new NotSupportedException("Skia must be the render backend for the FAColorPicker");

using var lease = leaseFeature.Lease();
var skDC = lease.SkCanvas;

// draw
```

What's the new way to do this? The closest I can find is to do a custom draw op that uses an ImmediateDrawingContext that I can query for Skia but that's annoying to implement and just adds unnecessary complexity here.

**Comment by kekekeks** [comment]
You can use WriteableBitmap and wrap it into an SKSurface instead.
